### PR TITLE
jsonschema: add "null" to `type` of non required property

### LINKED ISSUE #453: Required enforced even when not asked
in the github.com/zero-os/0-metadata/ project we have discovered that the field posix in the dir type on the github.com/zero-os/0-metadata/api_spec/main.raml is marked as "required: false" but the code generated complains when such field is not available.

In github.com/zero-os/0-metadata/tests/test_dir.py you can see that the test fails when the posix field is deleted in the creation of the dir object.

It fails the json validation even when not required.

### PR DESCRIPTION
**Pull request by iwanbk**
Fixes #453

### CODE REVIEW

File: raml/jsonschema.go [hunk 1]
```diff
@@ -236,6 +237,8 @@ func newProperty(rp Property) property {
 	if rp.IsArray() && !rp.IsBidimensiArray() {
 		p.Type = "array"
 		p.Items = newArrayItem(rp.ArrayType())
```
Comment (serboctor) [inline raml/jsonschema.go]: should we also support `null` for non-required arrays? or `[]` is enough?

Comment (iwanbk) [reply raml/jsonschema.go]: > should we also support null for non-required arrays

I was sure that there was no problem with array, but just tested it again, it has same bug.
need to investigate it

>  or [] is enough?

What do you mean? I don't get it

Comment (serboctor) [reply raml/jsonschema.go]: I mean do we want to support none too or forcing the user to send an empty list is enough.

Comment (iwanbk) [reply raml/jsonschema.go]: > I mean do we want to support none too

I think we have to support too, for these reasons:
- the generated class doesn't produce `[]`
- to stay consistent

Comment (iwanbk) [reply raml/jsonschema.go]: i've added the support

**Pull request merged**